ocfweb/docs/docs/staff/backend/libvirt.md

```diff
@@ -0,0 +1,128 @@
+`/dev/vg/<vm-name>`.
+
+[qemu]: https://www.qemu.org/
+[kvm]: https://www.linux-kvm.org/
+[libvirt]: https://libvirt.org/
+
+
+## Administration
+
+### How do I view which VMs are on a hypervisor?
+
+SSH into the hypervisor, then run `sudo virsh list --all`:
+
+    kpengboy@hal:~$ sudo virsh list --all
+     Id    Name                           State
+    ----------------------------------------------------
+     1     fallingrocks                   running
+     3     limniceruption                 running
+     4     spriggs                        running
+     5     dev-flood                      running
+     11    dev-whiteout                   running
+     15    dev-death                      running
+     18    dev-tsunami                    running
+     -     dev-anthrax                    shut off
+     -     dev-pestilence                 shut off
+     -     dev-werewolves                 shut off
+     -     zombies                        shut off
+
+### How do I turn on a VM?
+
+On the hypervisor, run `sudo virsh start <vm-name>`.
+
+### How do I turn off a VM?
+
+You can SSH into the VM and run the `shutdown` command, or you can run
+`sudo virsh stop <vm-name>` on the hypervisor which hosts it.
+
+If it's a public-facing VM (e.g. tsunami), remember to give a positive amount
+of time to the shutdown command, so users have adequate warning.
+
+### How do I make a VM automatically turn on when the hypervisor boots?
+
+On the hypervisor, run `sudo virsh autostart <vm-name>`.
+
+You can list which VMs are set to autostart with `sudo virsh list --all
+--autostart`.
+
+Firestorm is set to autostart because it must be running in order for any staff
+to log in (other than by using the root account). Other VMs are not set to
+autostart because if they start before LDAP and Kerberos are available, logins
+may not necessarily work properly.
+
+### Is there a GUI for all of this?
+
+You can run virt-manager on the hypervisors to graphically start and stop VMs,
+view their virtual monitors, and do many other similar things. To access
+virt-manager, SSH into the hypervisor with X forwarding (`-X`) enabled. Then, on
+the hypervisor, run `sudo XAUTHORITY=~/.Xauthority virt-manager`.
+
+#### If you just want to see a single VM's virtual monitor
+
+You can use the virt-viewer tool. Replace `virt-manager` in the above command
+with `virt-viewer <vm-name>`.
+
+Alternatively, you can directly connect to the VM's VNC display. This procedure
+gives much better performance than X forwarding, especially outside of the OCF
+network.  To do this, first run `sudo virsh vncdisplay <vm-name>` on the
+hypervisor, and record what it prints out. Then, from your local computer, run
+`vncviewer -via <hypervisor> <output of virsh vncdisplay>`. So for instance, if
+`virsh vncdisplay` printed out `127.0.0.1:10`, and the hypervisor were jaws,
+then you would run `vncviewer -via jaws 127.0.0.1:10` (or `localhost:10`).
```

Comment:
@jvperrin any comments?

Reply:
Looks fine, I generally just use a SSH tunnel instead of using `-via`, but this seems even easier since it's just the one command. Thanks!